scripts/workers/code_upload_worker_utils/install_dependencies.sh

```
@@ -50,11 +59,100 @@ kubectl apply -f /code/scripts/workers/code_upload_worker_utils/persistent_volum
+# from silently attempting an unsupported jump. Upgrade Cilium manually,
+# one minor version at a time, before advancing this pin.
+#
+# `helm list` failing (API unreachable, RBAC, corrupted Helm storage) and
+# "no release found" both leave the command substitution empty - they must
+# not be treated the same way. An empty result from a *failed* lookup would
+# silently skip this guard and fall through to an unattended install with no
+# idea whether an incompatible version is already there, defeating the point
+# of the check. Only an empty result from a lookup that *succeeded* means
+# "no release yet, safe to proceed."
+helm_list_output="$(helm list --namespace kube-system --filter '^cilium$' --output json 2>&1)"
+if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)